### PR TITLE
[307] Avoid NPEs if Session::close is called on a closed session

### PR DESCRIPTION
**Pull request by mPorhel**
Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/307

### CODE REVIEW

File: plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/session/SampleSessionTest.java [hunk 1]
```diff
@@ -248,6 +248,45 @@ public void testDeleteSessionWithRepresentationOpened() throws Exception {
...
+        assertFalse("UI Session should no more be opened.", sessionUI.isOpen());
+        sessionUI = SessionUIManager.INSTANCE.getUISession(session);
+        assertNull("UI Session should no more exists.", sessionUI);
+
+        for (final Session managerSession : new ArrayList<Session>(SessionManager.INSTANCE.getSessions())) {
+            if (managerSession == session) {
+            }
+        }
+
+        // Check that a second session.close() call does not trigger errors.
```
Comment (pcdavid) [inline plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/session/SampleSessionTest.java:286]: Unfinished?

File: plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java [hunk 2]
```diff
@@ -1360,9 +1368,11 @@ public void close(IProgressMonitor monitor) {
             broker = null;
         }
 
-        CommandStackUtil.flushOperations(transactionalEditingDomain.getCommandStack());
-        // Unload all referenced resources
-        unloadAllResources(monitor);
+        if (transactionalEditingDomain != null) {
```
Comment (pcdavid) [inline plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java:1371]: Why not use `ted` defined above?

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java (hunk 2 above)
Comment (pcdavid) [reply plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java:1371]: Discussed with @mPorhel: this is on purpose to minimize the amount of code changed this close to the 7.4.0 release.

File: plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/session/SampleSessionTest.java (hunk 1 above)
Comment (mPorhel) [reply plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/session/SampleSessionTest.java:286]: Still in my staging, it seems that I pushed my last version after forgotting to commit/ammend it.

File: plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java (hunk 2 above)
Comment (mPorhel) [reply plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/session/danalysis/DAnalysisSessionImpl.java:1371]: I confirm, we should rationalize the method and get the ted in the same way every way, but so close to the release, I tried to minimize the changes

**Review by pcdavid — APPROVED** [review]

**Pull request merged**